src/cargo/core/compiler/mod.rs

```diff
@@ -485,16 +485,34 @@ fn rustc(
         target: &Target,
         current_id: PackageId,
     ) -> CargoResult<()> {
+        let mut library_paths = vec![];
+
+        for key in build_scripts.to_link.iter() {
+            let output = build_script_outputs.get(key.1).ok_or_else(|| {
+                internal(format!(
+                    "couldn't find build script output for {}/{}",
+                    key.0, key.1
+                ))
+            })?;
+            library_paths.extend(output.library_paths.iter());
+        }
+
+        library_paths.sort_by_key(|p| match p {
```

Comment:
We've implemented (or derived) Ord, so `sort` should be good enough. No need to

```suggestion
        library_paths.sort()
```

Reply:
Unfortunately not. This does a grouped sort, keeping the relative ordering within a group the same (i.e. it moves all Cargo artifacts to the front of the search path and external paths to the end, but within cargo artifacts & within external paths the relative order of specified paths doesn't change. If you just did .sort, then it would sort by path within the group & screw up the ordering that build.rs requested.

Reply:
You're correct. I somehow thought it was already handled as I saw the manual `Ord` impl.